src/bundlers/node/typeCheckerBundler.ts

```diff
@@ -0,0 +1,62 @@
+import { BundlerInput, BundlerInterface, BundlerOutput } from "../bundler.interface";
+import ts from "typescript";
+import log from "loglevel";
+import fs from "fs";
+import { tsconfig } from "../../utils/configs";
+import path from "path";
+import { writeToFile } from "../../utils/file";
+import { debugLogger } from "../../utils/logging";
+
+export class TypeCheckerBundler implements BundlerInterface {
+    // Call this class only once
+    static used = false;
+
+    #generateTsconfigJson() {
+        if (fs.existsSync("tsconfig.json")) {
+            return;
+        } else {
+            console.log("No tsconfig.json file found. We will create one...")
+            tsconfig.compilerOptions.rootDir = ".";
+            tsconfig.compilerOptions.outDir = path.join(".", "build");
+            tsconfig.include = [path.join(".", "**/*")];
+            writeToFile(process.cwd(), "tsconfig.json", JSON.stringify(tsconfig, null, 4));
+        }
+    }
+
+    static logChangeDetection() {
+        TypeCheckerBundler.used = false;
+    }
+
+    async bundle(input: BundlerInput): Promise<BundlerOutput> {
+        if (TypeCheckerBundler.used) {
+            return input;
+        }
+        TypeCheckerBundler.used = true;
```

Comment:
Interesting solution. However, while testing, I've noticed that type checker is executed only once for `genezio local`.  If I save the type checker is not executed again. However, I think we want this behaviour. Type checking every time makes `genezio local` very slow. I've noticed the `logChangeDetection` function getting call in local.ts which makes me think that maybe this was not your intention.

Reply:
I have removed the type check on `local` as it was consuming a lot of time.